### PR TITLE
util/client.go: add method to specify pod security admission level

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @stlaz @tkashem @soltysh @deads2k

### CODE REVIEW

**Comment by stlaz** [comment]
/retest
infra issues

File: go.mod [hunk 1]
```diff
@@ -72,6 +72,7 @@ require (
 	k8s.io/kubelet v0.23.0
 	k8s.io/kubernetes v1.23.0
 	k8s.io/legacy-cloud-providers v0.23.0
+	k8s.io/pod-security-admission v0.0.0
```
Comment (stlaz) [inline go.mod:75]: That's an odd version. Should be at least v0.23.0.

Comment (s-urbaniak) [reply go.mod:75]: indeed, this is what `go mod tidy` generated. let me try to fix this manually.

**Comment by s-urbaniak** [comment]
holding to verify e2e tests

**Comment by s-urbaniak** [comment]
This blocked until we have the newest k8s bump available in origin. The bump was reverted in https://github.com/openshift/origin/pull/26939 because SDN e2e tests broke. There is work in progress in https://github.com/openshift/kubernetes/pull/1228 to unskip those tests. Once merged we can continue here.

File: test/extended/util/client.go [hunk 2]
```diff
@@ -105,8 +106,17 @@ func NewCLIWithFramework(kubeFramework *framework.Framework) *CLI {
 	return cli
 }
 
+// NewCLIWithPodSecurityEnforceLevel initializes the CLI the same way as `NewCLI()`
+// but the given pod security level is applied to the created e2e test namespace.
+func NewCLIWithPodSecurityEnforceLevel(project string, level admissionapi.Level) *CLI {
```
Comment (deads2k) [inline test/extended/util/client.go:111]: non-binding, but if it were me, I would have three methods: `NewCLIWithRestrictedNamespace`, `NewCLIWithBaselineNamespace`, `NewCLIWithPrivilegedNamespace`

**Comment by s-urbaniak** [comment]
blocked by https://github.com/openshift/origin/pull/26955

File: test/extended/util/client.go (hunk 2 above)
Comment (stlaz) [inline test/extended/util/client.go:111]: this should now probably be called just `NewCLIWithPodSecurityLevel` (remove the `Enforce` from the name)

Comment (s-urbaniak) [reply test/extended/util/client.go]: fixed

**Comment by s-urbaniak** [comment]
let's hold this as we pursue a different strategy when this was approved:

- default in this PR to `privileged` for all e2e tests
- explicitly set to `restricted` in a subsequent PR

This reduces the risk of breaking existing e2e tests and allows us to iteratively fix them.

/cc @deads2k @soltysh

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by stlaz** [comment]
/retest-required
irrelevant test failures

File: test/extended/util/client.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"k8s.io/client-go/util/retry"
```
Comment (stlaz) [inline test/extended/util/client.go:11]: order!

Comment (s-urbaniak) [reply test/extended/util/client.go:11]: let's see how CI goes, i don't want to risk another e2e failure rush. i'll address in a followup

**Pull request merged**